ethereum_history_api/oracles/src/noir/noir_js/encode.ts

```diff
@@ -1,8 +1,19 @@
-import { isHex } from 'viem';
+import { Hex, isHex } from 'viem';
 import { BYTE_HEX_LENGTH } from '../../util/const.js';
 
 // ENCODERS
-export function encodeHexString(value: string): Uint8Array {
+export function encodeHexString(value: string): Hex[] {
+  if (!isHex(value)) {
+    throw new Error(`Invalid hexstring: ${value}`);
+  }
+  const chunks: Hex[] = [];
+  for (let i = 2; i < value.length; i += BYTE_HEX_LENGTH) {
+    chunks.push(`0x${value.substr(i, BYTE_HEX_LENGTH)}`);
+  }
+  return chunks;
+}
```

Comment:
Seems like we can reuse encodeHexStringToArray here and have less code in this function

Reply:
At first I implemented it as you're mentioning but then I thought it's an overkill to first convert a hex value like '0x1234' to array of integers [18, 52], only to then convert it to array of hexes ['0x12', '0x34']. What do you think?

Reply:
Please check the new version of this function. I changed it from encodeField to converting ints by hand, because encodeField converted 0 t0 '0x' instead of '0x00' which was not desired.